src/main/scala/zio/telemetry/Telemetry.scala

```diff
@@ -0,0 +1,213 @@
+package zio.telemetry
+
+import io.opentracing.Span
+import io.opentracing.SpanContext
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.tag.Tag
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters._
+import zio.Exit
+import zio.FiberRef
+import zio.IO
+import zio.Task
+import zio.UIO
+import zio.ZIO
+import zio.ZManaged
+import zio.clock.Clock
+
+trait Telemetry extends Serializable {
+  def telemetry: Telemetry.Service
+}
+
+object Telemetry {
+
+  trait Service {
+    def spanRef: FiberRef[Span]
```

Comment:
Perhaps rename to `currentSpan`.

Reply:
With fold we need type parameters because they are not inferred correctly. I prefer pattern matching over explicit type parameters. :)